src/libraries/System.Net.Quic/src/System/Net/Quic/QuicListener.cs

```diff
@@ -165,14 +168,7 @@ public async ValueTask<QuicConnection> AcceptConnectionAsync(CancellationToken c
                 PendingConnection pendingConnection = await _acceptQueue.Reader.ReadAsync(cancellationToken).ConfigureAwait(false);
                 await using (pendingConnection.ConfigureAwait(false))
                 {
-                    QuicConnection? connection = await pendingConnection.FinishHandshakeAsync(cancellationToken).ConfigureAwait(false);
-                    // Handshake failed, discard this connection and try to get another from the queue.
-                    if (connection is null)
-                    {
-                        continue;
-                    }
-
-                    return connection;
+                    return await pendingConnection.FinishHandshakeAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
This no longer needs to be inside a while loop, right?

Reply:
Fixed.